.github/workflows/rust.yml

```diff
@@ -1,46 +1,67 @@
-name: Build
+name: ion-rust CI Build
 
 on: [push]
 
 jobs:
-  linux:
-    runs-on: ubuntu-latest
+  build:
+    name: Build and Test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
 
     steps:
-      - name: Checkout
+      - name: Env Variable Setup 
+        if: matrix.os == 'windows-latest'
+        run: | 
+          echo '::set-env name=LIBCLANG_PATH::C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\Llvm\x64\bin'
+      - name: Git Checkout
         uses: actions/checkout@v2
         with:
           submodules: recursive
-      - name: Build
-        run: cargo build --verbose --workspace
-      - name: Tests
-        run: cargo test --verbose --workspace
-
-  macos:
-    runs-on: macos-latest
-
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
+      - name: Rust Toolchain
+        uses: actions-rs/toolchain@v1
         with:
-          submodules: recursive
-      - name: Build
-        run: cargo build --verbose --workspace
-      - name: Tests
-        run: cargo test --verbose --workspace
-
-  windows:
-    runs-on: windows-latest
-    env:
-      # needed for bindgen
-      LIBCLANG_PATH: C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\Llvm\x64\bin
+          profile: minimal
+          toolchain: stable
+          override: true
```

Comment:
Is this specifically to choose the tool chain versus what is installed in the base image?

Reply:
Yes. While trying grcov I needed nightly and this was the GitHub Action that I found which allows you to pick Rust versions as well as (when we define a minimum version) a minimum version feature to build and run.